homeassistant/components/homewizard/config_flow.py

```diff
@@ -72,25 +73,25 @@ async def async_step_user(
                 errors=None,
             )
 
-        error = await self._async_try_connect(user_input[CONF_IP_ADDRESS])
-        if error is not None:
+        # Fetch device information
+        device_info_or_error = await self._async_try_connect(
```

Comment:
I read the additional PR feedback as this should raise an exception, e.g. `raise FormError("api_not_enabled")` rather than returning a string

Reply:
Nice idea, not considered. Works really well and changes are pushed with latest commit!